scipy/spatial/distance.py

```diff
@@ -2648,34 +2655,25 @@ def is_valid_y(y, warning=False, throw=False, name=None):
     False
 
     """
-    y = np.asarray(y, order='c')
-    valid = True
+    y = _asarray(y)
```

Comment:
This is a regression as y can legitimately be a list of arbitrary python objects, which causes `_asarray` to raise.
Working on a fix.